src/utils/token_fiat_price.ts

```diff
@@ -0,0 +1,17 @@
+import type { FiatID, TokenID } from './fiat_oracle_types';
+
+/**
+ * Immutable data container class representing a market price in USD for a particular AR amount
+ * that enforces valid number ranges for AR and USD price values.
+ */
+export class TokenToFiatPrice {
+	/**
+	 * @returns an ARFiatPrice instance with the given AR and USD amount
+	 * @throws {@link Error} if negative or non-integer values are provided for the fiat value
+	 */
+	constructor(readonly fiat: FiatID = 'usd', readonly token: TokenID = 'arweave', readonly amount: number) {
+		if (amount < 0 || !Number.isInteger(amount)) {
+			throw new Error(`amount (${amount}) should be a non-negative integer value.`);
```

Comment:
I don't think we can presume that any fiat or token types are non-divisible, at least not at this level of the architecture.

Reply:
Hmmm how is that? Do you mean impossible pairs?

Reply:
I think he means, the rates can be decimals.

I tested the latest updates, and that is the case here with usd at `19.2`:

```
Failed to use fiat oracle: Error: rate (19.2) should be a non-negative integer value.
    at new TokenFiatPair (token_fiat_price.ts:17)
    at TokenToFiatOracle.updateCachePair (token_to_fiat_oracle.ts:106)
    at token_to_fiat_oracle.ts:96
    at Array.forEach (<anonymous>)
    at token_to_fiat_oracle.ts:94
    at Array.forEach (<anonymous>)
    at TokenToFiatOracle.fetchPrices (token_to_fiat_oracle.ts:93)
    at async TokenToFiatOracle.checkCache (token_to_fiat_oracle.ts:84)
    at async TokenToFiatOracle.getPriceForFiatTokenPair (token_to_fiat_oracle.ts:68)
    at async getFiatToken (useFiatOracle.ts:10)
```

Reply:
Right, so it should be nice now: 0ddaefe124f96209e6583530ac422593f318292a